### PR TITLE
pythonPackages.caiman: init at 1.6.4

### PR DESCRIPTION
**Pull request by tbenst**
###### Motivation for this change
Add CaImAn, a Python toolbox for large scale Calcium Imaging data Analysis and behavioral analysis.

The current derivation segfaults, possibly due to improper use of `qt5.mkDerivationWith`

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by tbenst** [comment]
@ttuegel let me know if you have any ideas? thanks!

File: lib/licenses.nix [hunk 1]
```diff
@@ -100,6 +100,11 @@ lib.mapAttrs (n: v: v // { shortName = n; }) {
     fullName = ''BSD 3-clause "New" or "Revised" License'';
   };
 
+  bsd3-lbnl = spdx {
+    spdxId = "BSD-3-Clause-LBNL";
+    fullName = ''Lawrence Berkeley National Labs BSD variant license'';
+  };
+
```
Comment (jonringer) [inline lib/licenses.nix]: This should be a separate commit

File: pkgs/development/python-modules/caiman/default.nix [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+}:
+
+qt5.mkDerivationWith buildPythonPackage rec {
+  pname = "caiman";
+  version = "1.6.4";
+
+  src = fetchFromGitHub {
+      owner = "flatironinstitute";
+      repo = pname;
+      rev = "v"+version;
```
Comment (jonringer) [inline pkgs/development/python-modules/caiman/default.nix]:
```suggestion
      rev = "v${version}";
```

File: pkgs/development/python-modules/hdmf/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  pname = "hdmf";
+  version = "1.3.3";
+  # pynwb depends on 1.3.3. 1.4 does not work
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0r3wj5pifhzfa773lnkl7garc18svfvpk5y5p0d0ww4xc2hab1k1";
+  };
+
+  checkInputs = [ unittest2 ];
```
Comment (jonringer) [inline pkgs/development/python-modules/hdmf/default.nix]: part of std library for python3
```suggestion
  checkInputs = lib.optionals isPy27 [ unittest2 ];
```

File: pkgs/development/python-modules/peakutils/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+    propagatedBuildInputs = [
+      numpy
+      scipy
+      pandas
+    ];
+
+  # https://bitbucket.org/lucashnegri/peakutils/issues/36/exp-not-found
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/peakutils/default.nix]: please try to run as many tests as possible, should have some unit tests

File: pkgs/development/python-modules/peakutils/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+, scipy
+, pandas
+}:
+
+buildPythonPackage rec {
+  pname = "peakutils";
+  version = "1.3.2";
+
+  src = fetchPypi {
+    pname = "PeakUtils";
```
Comment (jonringer) [inline pkgs/development/python-modules/peakutils/default.nix]: URLs are case insenstive
```suggestion
```

File: pkgs/development/python-modules/peakutils/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    inherit version;
```
Comment (jonringer) [inline pkgs/development/python-modules/peakutils/default.nix]:
```suggestion
    inherit version pname;
```

File: pkgs/development/python-modules/pynwb/default.nix [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      --replace "python-dateutil==2.8.0" "python-dateutil" \
+      --replace "ruamel.yaml==0.16.5" "ruamel.yaml" \
+      --replace "six==1.12.0" "six"
+    rm tests/build_fake_data.py
+    rm tests/build_fake_extension.py
+  '';
+
+  # doCheck = false;
+
+  checkInputs = [ unittest2 ];
```
Comment (jonringer) [inline pkgs/development/python-modules/pynwb/default.nix]:
```suggestion
  checkInputs = lib.optionals isPy27 [ unittest2 ];
```

**Comment by onny** [comment]
Please rebase

**Comment by Mic92** [comment]
Closing because of lack of activity from the author. Please re-open if you plan to work on this.
